test/views/UploadListPageSpec.scala

```diff
@@ -57,34 +74,68 @@ class UploadListPageSpec extends SpecBase with ViewBehaviours {
-      Selectors.labelForRadioButton(1) -> yesOption,
-      Selectors.labelForRadioButton(2) -> noOption,
-      Selectors.label -> uploadAnotherFile,
-      Selectors.button -> continueButton
-    )
-    behave like pageWithExpectedMessages(expectedContent)
-  }
+    "return single file header and title added" when {
+      def applyView(form: Form[_]): HtmlFormat.Appendable = uploadListPage.apply(
+        form, radioOptions, controllers.routes.OtherReasonController.onPageLoadForUploadComplete(NormalMode), uploadSingleRow, None)(userRequestWithCorrectKeys, messages, appConfig)
+
+      implicit val doc: Document = asDocument(applyView(formProvider))
 
-  "return max of 5 files added" when {
-    def applyView(form: Form[_]): HtmlFormat.Appendable = uploadListPage.apply(
-      form, radioOptions, controllers.routes.OtherReasonController.onPageLoadForUploadComplete(NormalMode), uploadMaxRow)(userRequestWithCorrectKeys,messages,appConfig)
-    implicit val doc: Document = asDocument(applyView(formProvider))
-
-    val expectedContent = Seq(
-      Selectors.title -> titleMaxFiles,
-      Selectors.h1 -> h1MaxFiles,
-      Selectors.button -> continueButton
-    )
-    behave like pageWithExpectedMessages(expectedContent)
+      val expectedContent = Seq(
+        Selectors.title -> titleSingleFile,
+        Selectors.h1 -> h1SingleFile,
+        Selectors.labelForRadioButton(1) -> yesOption,
+        Selectors.labelForRadioButton(2) -> noOption,
+        Selectors.label -> uploadAnotherFile,
+        Selectors.button -> continueButton
+      )
+      behave like pageWithExpectedMessages(expectedContent)
+    }
+
+    "return max of 5 files added" when {
+      def applyView(form: Form[_]): HtmlFormat.Appendable = uploadListPage.apply(
+        form, radioOptions, controllers.routes.OtherReasonController.onPageLoadForUploadComplete(NormalMode), uploadMaxRow, None)(userRequestWithCorrectKeys, messages, appConfig)
+
+      implicit val doc: Document = asDocument(applyView(formProvider))
+
+      val expectedContent = Seq(
+        Selectors.title -> titleMaxFiles,
+        Selectors.h1 -> h1MaxFiles,
+        Selectors.button -> continueButton
+      )
+      behave like pageWithExpectedMessages(expectedContent)
+    }
+
+    "show the single duplicate insetText when a user uploads a duplicate document" when {
+      def applyView(form: Form[_]): HtmlFormat.Appendable = uploadListPage.apply(
+        form, radioOptions, controllers.routes.OtherReasonController.onPageLoadForUploadComplete(NormalMode), uploadMaxRow, optInsetText("1", "2"))(userRequestWithCorrectKeys, messages, appConfig)
+
+      implicit val doc: Document = asDocument(applyView(formProvider))
+
+      val expectedContent = Seq(
+        Selectors.title -> titleMaxFiles,
+        Selectors.h1 -> h1MaxFiles,
+        Selectors.insetText -> insetTextMsg("1", "2"),
+        Selectors.button -> continueButton
+      )
+
+       behave like pageWithExpectedMessages(expectedContent)
+    }
```

Comment:
Could we have a test for multiple duplicates uploaded of the same set i.e. 
Document 1 has the same contents as Document 2, 3 and 4. You can remove duplicate documents using the ‘Remove‘ link.